controller/workitem_whitebox_test.go

```diff
@@ -164,6 +164,59 @@ func (rest *TestWorkItemREST) TestConvertJSONAPIToWorkItemWithLegacyDescription(
 
 }
 
+// TODO(Ibrahim) - This test should be removed after we remove support for system.* fields
+func (rest *TestWorkItemREST) TestConvertJSONAPIToWorkItemWithDotAndPeriodFields() {
+	resource.Require(rest.T(), resource.Database)
+	//given
+	fxt := tf.NewTestFixture(rest.T(), rest.DB, tf.CreateWorkItemEnvironment())
+	rest.T().Run("system.* to system_*", func(t *testing.T) {
+		attributes := map[string]interface{}{
+			"system.title":   "title",
+			"system.order":   1,
+			"system.foo.bar": "helloworld",
+			"some.foo.bar":   "xyz",
+		}
+		source := prepareWI2(attributes, fxt.WorkItemTypes[0].ID, fxt.Spaces[0].ID)
+		target := &workitem.WorkItem{Fields: map[string]interface{}{}}
+		err := application.Transactional(rest.GormDB, func(app application.Application) error {
+			return ConvertJSONAPIToWorkItem(context.Background(), "", app, source, target, fxt.WorkItemTypes[0].ID, fxt.Spaces[0].ID)
+		})
+		require.NoError(t, err)
+		require.NotNil(t, target)
+		require.NotNil(t, target.Fields)
+		// Check if new field names are present and have correct value
+		assert.Equal(t, target.Fields[workitem.SystemTitle], "title")
+		assert.Equal(t, target.Fields[workitem.SystemOrder], 1)
+
+		// Check if old field name is removed
+		assert.Nil(t, target.Fields["system.title"])
```

Comment:
I think things become clearer now. You're testing the conversion from JSONAPI to the internal representation and there you can remove the old fields. That makes sense. Please confirm.

Reply:
Yes. The result that comes out of `ConvertJSONAPIToWorkitem` will never have any `system.` field. We clean up the payload and `add/remove` fields so that the resulting workitem has only `system_` fields without any loss of information.